pepr/robustness/foolbox_wrapper.py

```diff
@@ -0,0 +1,3230 @@
+            "SaltAndPepperNoiseAttack",
+        )
+
+
+class LinearSearchBlendedUniformNoiseAttack(BaseAttack):
+    """
+    foolbox.attacks.LinearSearchBlendedUniformNoiseAttack wrapper class.
+
+    Attack description:
+    Blends the input with a uniform noise input until it is misclassified.
+
+    Parameters
+    ----------
+    attack_alias : str
+        Alias for a specific instantiation of the class.
+    attack_pars : dict
+        Dictionary containing all needed attack parameters:
+
+        * distance (foolbox.distances.Distance): Distance measure for which minimal
+          adversarial examples are searched.
+        * directions (int): (optional) Number of random directions in which the
+          perturbation is searched.
+        * steps (int): (optional) Number of blending steps between the original image
+          and the random directions.
+        * epsilons (list): List of one or more Epsilons for the attack.
+
+    data : numpy.ndarray
+        Dataset with all input images used to attack the target models.
+    labels : numpy.ndarray
+        Array of all labels used to attack the target models.
+    data_conf : dict
+        Dictionary describing for every target model which record-indices should be used
+        for the attack.
+
+        * attack_indices_per_target (numpy.ndarray): Array of indices of images to
+          attack per target model.
+
+    target_models : iterable
+        List of target models which should be tested.
+    """
+
+    def __init__(
+        self, attack_alias, attack_pars, data, labels, data_conf, target_models
+    ):
+        # Handle specific attack class parameters
+        params = {}
+        if "directions" in attack_pars:
+            params["directions"] = attack_pars["directions"]
+        if "steps" in attack_pars:
+            params["steps"] = attack_pars["steps"]
+        foolbox_attack = fb.attacks.LinearSearchBlendedUniformNoiseAttack(
+            distance=attack_pars["distance"], **params
+        )
+
+        pars_descriptors = {
+            "distance": "Distance",
+            "directions": "Random directions",
+            "steps": "Steps",
+        }
+
+        super().__init__(
+            attack_alias,
+            attack_pars["epsilons"],
+            data,
+            labels,
+            data_conf["attack_indices_per_target"],
+            target_models,
+            foolbox_attack,
+            pars_descriptors,
+        )
+
+        self.report_section = report.ReportSection(
+            "Linear Search Blended Uniform Noise Attack",
+            self.attack_alias,
+            "LinearSearchBlendedUniformNoiseAttack",
+        )
+
+
+class BinarizationRefinementAttack(BaseAttack):
+    """
+    foolbox.attacks.BinarizationRefinementAttack wrapper class.
+
+    Attack description:
+    For models that preprocess their inputs by binarizing the inputs, this attack can
+    improve adversarials found by other attacks. It does this by utilizing information
```

Comment:
Probably better "... improve adversarial examples..."?

Reply:
This is copied from the official foolbox documentation. I would keep it like that.